src/main/kotlin/com/demonwav/mcdev/platform/sponge/inspection/SpongeInvalidGetterTargetInspection.kt

```diff
@@ -0,0 +1,106 @@
+import com.siyeh.ig.BaseInspection
+import com.siyeh.ig.BaseInspectionVisitor
+import com.siyeh.ig.InspectionGadgetsFix
+import kotlin.concurrent.thread
+
+class SpongeInvalidGetterTargetInspection : BaseInspection() {
+
+    override fun getDisplayName() = "@Getter targeted method does not exist"
+
+    override fun buildErrorString(vararg infos: Any?) = staticDescription
+
+    override fun getStaticDescription() = "@Getter must target a method accessible from the event class of this listener"
+
+    override fun buildVisitor(): BaseInspectionVisitor {
+        return object : BaseInspectionVisitor() {
+            override fun visitMethod(method: PsiMethod) {
+                if (method.parameters.size < 2 || !method.isValidSpongeListener()) {
+                    return
+                }
+
+                for (i in 1 until method.parameters.size) {
+                    val parameter = method.parameters[i]
+                    val getterAnnotation = parameter.getAnnotation(SpongeConstants.GETTER_ANNOTATION) as? PsiAnnotation ?: continue
+                    val getterTarget = getterAnnotation.resolveSpongeGetterTarget()
+                    if (getterTarget == null) {
+                        val attribute = getterAnnotation.findAttributeValue("value")
+                        if (attribute == null) {
+                            registerError(getterAnnotation, getterAnnotation)
+                        } else {
+                            registerError(attribute)
+                        }
+                        continue
+                    }
+                }
+            }
+        }
+    }
+
+    override fun buildFix(vararg infos: Any?): InspectionGadgetsFix? {
+        if (infos.isEmpty()) {
+            return null
+        }
+
+        val annotation = infos[0] as PsiAnnotation
+        if (!annotation.isWritable) {
+            return null
+        }
+
+        return QuickFix(annotation, "Add target method")
+    }
+
+    class QuickFix(annotation: PsiAnnotation, private val name: String) : InspectionGadgetsFix() {
+
+        private val pointer = SmartPsiPointer(annotation)
+
+        override fun doFix(project: Project, descriptor: ProblemDescriptor?) {
+            doFix(project, pointer.element as PsiAnnotation)
+        }
+
+        override fun getFamilyName() = name
+
+        override fun getName() = name
+
+        companion object {
+
+            fun doFix(project: Project, annotation: PsiAnnotation) {
+                val value = JavaPsiFacade.getElementFactory(project).createExpressionFromText("\"\"", annotation)
+                val newValue = annotation.setDeclaredAttributeValue(null, value)
+                val editor = PsiEditorUtil.Service.getInstance().findEditorByPsiElement(annotation) ?: return
+                editor.caretModel.removeSecondaryCarets()
+                editor.selectionModel.removeSelection()
+                editor.caretModel.moveToOffset(newValue.textOffset + 1)
+                thread {
```

Comment:
What are you trying to do here? Calling `invokeLater` immediately in a new `thread` doesn't really make any sense.

Reply:
It's been a while so I can't remember what I tried to do to end up with this, but I agree this is nonsense, I'll try to keep the same behavior with something better.